test/libsolidity/LibSolc.cpp

```diff
@@ -62,7 +62,7 @@ Json::Value compile(string const& _input, CStyleReadFileCallback _callback = nul
 	string output(solidity_compile(_input.c_str(), _callback, nullptr));
 	Json::Value ret;
 	BOOST_REQUIRE(jsonParseStrict(output, ret));
-	solidity_free();
+	solidity_reset();
```

Comment:
Perhaps move this after compile? Since we copy it into a string anyway.

Reply:
So you mean out of `compile()` and then called after each call to `compile()`?

Reply:
I mean just to move it 2 lines above.